### PR TITLE
More automation progress

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
This PR does a bunch of things aimed at making the output of CI suitable for upload to hackage. It's not the whole story across all of our builds but it's good progress. I also take the chance here to stop generating a near useless exe in `ghc-lib` cabal.

### CODE REVIEW

File: CI.hs [hunk 1]
```diff
@@ -104,10 +115,28 @@ main = do
...
+rmFiles fs = forM_ fs $
+  \f -> removeFile f `catchIOError` (const  $ return ())
+
+rmDirs :: [FilePath] -> IO ()
+rmDirs ds = forM_ ds $
+ \d -> removeDirectoryRecursive d `catchIOError` (const $ return ())
+
+tag :: IO String
+tag = do
+  DateTime y m d _ _ _ <- getCurrentDateTime
```
Comment (cocreature) [inline CI.hs]: Unless there is a specific reason for why we need to use the `dates` package I would recommend to use the much more popular `time` package which is even shipped with GHC.

File: CI.hs [hunk 2]
```diff
@@ -104,10 +115,28 @@ main = do
         cmd "stack sdist ghc --tar-dir=."
         cmd $ "tar -xvf " ++ target ++ ".tar.gz"
 
-      rmFiles :: [FilePath] -> IO ()
-      rmFiles fs = forM_ fs $
-        \f -> removeFile f `catchIOError` (const  $ return ())
+rmFiles :: [FilePath] -> IO ()
+rmFiles fs = forM_ fs $
+  \f -> removeFile f `catchIOError` (const  $ return ())
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply CI.hs]: ✔️

File: CI.hs [hunk 3]
```diff
@@ -104,10 +115,28 @@ main = do
...
+
+rmDirs :: [FilePath] -> IO ()
+rmDirs ds = forM_ ds $
+ \d -> removeDirectoryRecursive d `catchIOError` (const $ return ())
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply CI.hs]: ✔️

File: CI.hs [hunk 4]
```diff
@@ -104,10 +115,28 @@ main = do
...
+
+tag :: IO String
+tag = do
+  DateTime y m d _ _ _ <- getCurrentDateTime
+  return $  "0." ++ show y ++ show_ m ++ show_ d
+  where show_ = printf "%02d"
+
+applyPatchVersion :: String -> FilePath -> IO ()
+applyPatchVersion version file = do
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply CI.hs]: ✔️

File: CI.hs (hunk 1 above)
Comment (shayne-fletcher-da) [reply CI.hs]: ✔️

**Comment by shayne-fletcher-da** [comment]
@cocreature awesome review. thanks!

**Pull request merged**